components/infobox/wikis/starcraft/infobox_team_custom.lua

```diff
@@ -0,0 +1,358 @@
+---
+-- @Liquipedia
+-- wiki=starcraft
+-- page=Module:Infobox/Team/Custom
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local Json = require('Module:Json')
+local Math = require('Module:Math')
+local Namespace = require('Module:Namespace')
+local RaceIcon = require('Module:RaceIcon')
+local String = require('Module:StringUtils')
+local Team = require('Module:Infobox/Team')
+local Variables = require('Module:Variables')
+
+local Injector = require('Module:Infobox/Widget/Injector')
+local Cell = require('Module:Infobox/Widget/Cell')
+local Title = require('Module:Infobox/Widget/Title')
+local Breakdown = require('Module:Infobox/Widget/Breakdown')
+
+local Condition = require('Module:Condition')
+
+local ConditionTree = Condition.Tree
+local ConditionNode = Condition.Node
+local Comparator = Condition.Comparator
+local BooleanOperator = Condition.BooleanOperator
+local ColumnName = Condition.ColumnName
+
+local doStore = true
+local pagename = mw.title.getCurrentTitle().prefixedText
+
+local CustomTeam = Class.new()
+
+local CustomInjector = Class.new(Injector)
+local _LANGUAGE = mw.language.new('en')
+
+local _earnings = 0
+local _earnings_by_players_while_on_team = 0
+local _EARNINGS_MODES = {team = 'team'}
+local _ALLOWED_PLACES = {'1', '2', '3', '4', '3-4'}
+local _DISCARD_PLACEMENT = 99
+local _MAXIMUM_NUMBER_OF_PLAYERS_IN_PLACEMENTS = 35
+local _PLAYER_EARNINGS_ABBREVIATION = '<abbr title="Earnings of players while on the team">Player earnings</abbr>'
+
+local _args
+
+function CustomTeam.run(frame)
+	local team = Team(frame)
+	_args = team.args
+
+	team.getWikiCategories = CustomTeam.getWikiCategories
+	team.addToLpdb = CustomTeam.addToLpdb
+	team.createWidgetInjector = CustomTeam.createWidgetInjector
+
+	return team:createInfobox(frame)
+end
+
+function CustomInjector:addCustomCells(widgets)
+	table.insert(widgets, Cell({
+		name = 'Gaming Director',
+		content = {_args['gaming director']}
+	}))
+	table.insert(widgets, Cell({
+		name = 'TLPD',
+		content = {_args.tlpd}
+	}))
+
+	return widgets
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'earnings' then
+		_earnings = CustomTeam.calculateEarnings(_args)
```

Comment:
I get that earningsOnTeam & Medals needs to be customized, but does the standard earnings need that as well?

Reply:
the general idea was to save runtime and do it all in one query instead of several

Reply:
Pretty sure it won't save runtime, as the default earnings method from commons will be executed first, before the parse.

Reply:
yeah i copied it from sc2, which was written before the earnings default stuff was added to commons

Reply:
just checked the values
calculateForTeam calculates wrong for sc & sc2 due to 2v2, 3v3, 4v4, archon stuff
we would need to add the overwrite of the earnings module on those wikis first

also the value displayed in the `Approx. Total Winnings` cell on sc/sc2 is just the earnings of the team in team events
the earnings players have in non team events go into the `Player earnings` cell

Reply:
Ok, leaving out of scope for now.